src/librustc_resolve/late.rs

```diff
@@ -474,7 +481,8 @@ impl<'a, 'ast> Visitor<'ast> for LateResolutionVisitor<'a, '_, 'ast> {
             // Bail if there's no body.
             FnKind::Fn(.., None) => return visit::walk_fn(self, fn_kind, sp),
             FnKind::Fn(FnCtxt::Free | FnCtxt::Foreign, ..) => FnItemRibKind,
-            FnKind::Fn(FnCtxt::Assoc(_), ..) | FnKind::Closure(..) => NormalRibKind,
+            FnKind::Fn(FnCtxt::Assoc(_), ..) => NormalRibKind,
+            FnKind::Closure(..) => ClosureOrAsyncRibKind,
```

Comment:
This is an impossible case, it's better to insert some kind of panic here.

Reply:
I'm not entirely sure which case you're referring to here - but assuming that it is the `FnKind::Closure(..)` arm, then making that a `bug!` immediately fails when trying to compile `core`. Did you intend the `FnKind::Fn(FnCtxt::Assoc(_), ..)` arm?

Reply:
No, I meant `Closure`.

Reply:
Actually, it may be reachable.
But then the `ClosureOrAsyncRibKind` is created twice.
Something is suspicious here anyway.

Reply:
I've not changed anything regarding this in the most recent commit.